ui/helpers/job.js

```diff
@@ -85,6 +85,29 @@ export const isUnclassifiedFailure = function isUnclassifiedFailure(job) {
   return thFailureResults.includes(job.result) && !isClassified(job);
 };
 
+export const isIntermittent = function isIntermittent(jobToEvaluate, group) {
+  if (jobToEvaluate.result !== 'testfailed') {
+    return false;
+  }
+
+  console.log(jobToEvaluate);
+  const failedJobTypeNames = new Set();
+  for (const job of group.jobs) {
+    if (job.result === 'testfailed') {
+      failedJobTypeNames.add(job.job_type_name);
+    }
+  }
+
+  const intermittentJobTypeNames = new Set();
+  for (const job of group.jobs) {
+    if (job.result === 'success' && failedJobTypeNames.has(job.job_type_name)) {
+      intermittentJobTypeNames.add(job.job_type_name);
+    }
+  }
```

Comment:
now these loops are executed for each query, that is redundant and it might impact the performance if there are many jobs that matches.
can we instead export a function that takes `group`, and calculate `intermittentJobTypeNames`, and return a new function that takes `jobToEvaluate` and tests with pre-calculated `intermittentJobTypeNames` for each call ?